src/main/java/io/strimzi/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -521,4 +537,41 @@ private void handleCommit(AsyncResult<Void> commitResult) {
             this.commitHandler.handle(commitResult);
         }
     }
+
+    private void handleManualRecords(KafkaConsumerRecords<K, V> record) {
+        if (this.manualRecordBatchHandler != null) {
+            this.manualRecordBatchHandler.handle(record);
+        }
+    }
+
+    private void handleCommitedOffsets(AsyncResult<Map<TopicPartition, io.vertx.kafka.client.consumer.OffsetAndMetadata>> offsetsData) {
+        if (this.commitOffsetsHandler != null) {
+            this.commitOffsetsHandler.handle(offsetsData);
+        }
+    }
+
+    protected void consume(Handler<KafkaConsumerRecords<K, V>> manualRecordBatchHandler) {
+        this.manualRecordBatchHandler = manualRecordBatchHandler;
+
+        this.consumer.poll(this.pollTimeOut, pollResult -> {
+            if (pollResult.succeeded()) {
+                if (pollResult.result().size() > 0) {
+
+                    if (this.manualRecordBatchHandler != null){
+                        this.handleManualRecords(pollResult.result());
+                    }
+                }
+            }
```

Comment:
what happens if it doesn't succeed? The HTTP client doesn't get any response. So I guess that the `manualRecordBatchHandler` should be an `Handler<AsyncResult<KafkaConsumer......>` so that the bridge is able to send a HTTP 500 error if it fails.